btrace-instr/src/main/java/org/openjdk/btrace/instr/ClassCache.java

```diff
@@ -58,26 +93,40 @@ public ClassInfo get(ClassLoader cl, String className) {
     return get(cl, new ClassName(className));
   }
 
-  synchronized ClassInfo get(ClassLoader cl, ClassName className) {
-    Map<ClassName, ClassInfo> infos = getInfos(cl);
+  ClassInfo get(ClassLoader cl, ClassName className) {
+    ConcurrentMap<ClassName, ClassInfo> infos = getInfos(cl);
 
-    ClassInfo ci = infos.get(className);
-    if (ci == null) {
-      ci = new ClassInfo(this, cl, className);
-      infos.put(className, ci);
-    }
-    return ci;
+    return infos.computeIfAbsent(className, k -> new ClassInfo(ClassCache.this, cl, k));
   }
 
-  private synchronized Map<ClassName, ClassInfo> getInfos(ClassLoader cl) {
+  ConcurrentMap<ClassName, ClassInfo> getInfos(ClassLoader cl) {
     if (cl == null) {
       return bootstrapInfos;
     }
-    Map<ClassName, ClassInfo> infos = cacheMap.computeIfAbsent(cl, k -> new HashMap<>(500));
+    boolean[] rslt = new boolean[] {false};
```

Comment:
Would AtomicBoolean be better

Reply:
This might be questionable - atomic boolean adds the overhead of CAS handling and since the value is updated and read from the same thread we don't need to enforce the visibility of the changed value and the final array might be fine.
TBH, not sure what the diff in the performance would be (if any) but I used to use this pattern when multithreaded access was not required.

Reply:
agreed, though i am more curious why boolean[] instead of boolean?